### PR TITLE
Fix #2: Budget accountant

### LINKED ISSUE #2: Budget accountant
In the context of differential privacy, "budget accountants" refer to mechanisms or tools used to track and manage the privacy budget in a differential privacy system. The privacy budget, often symbolized as `ε (epsilon)`, is a key concept in differential privacy. It quantifies how much privacy loss is incurred by the release of data or data queries.

In differential privacy libraries like OpenDP, SmartNoise, Opacus, and diffprivlib, budget accountants are implemented as tools or functions that help keep track of the cumulative privacy budget.

We strongly recommend reading an introduction to [privacy budgeting](https://www.notion.so/oblv/Privacy-tracking-157bb6f41dc04e41aaeb43fa89acc6bd) before tackling this issue.

Following enhancements need to be added to the budget accountant:

- [ ] Multiple datasets must be supported, and the accountant must keep track of how much ε (and δ) is spent by each dataset. `dataset_id`, a very common field in most of the datasets can be used for this.
- [ ] Support for datasets which are related to each other. `related_dataset_ids` field can be used for this.
- [ ] Accountant can add to budget in the following ways - 
     - [ ] `add_to_budget(dataset_id / ids, epsilon, delta)`
     - [ ] `add_to_budget(dataset_id / ids, opendp_mechanism)`
You can read about what mechanisms are and how to extract ε (and δ) from [opendp docs](https://docs.opendp.org/en/stable/user/first-look-at-DP.html)
- [ ] The budget accountant must be as modular as possible, which can allow for different ways of handling different datasets. For example, if two datasets are partially correlated, total epsilon can be a mathematical function of the epsilon spent.

### PR DESCRIPTION
**Pull request by Dhruv127**

### CODE REVIEW

**Review by hardik-kapoor — DISMISSED** [review]

**Review by hardik-kapoor — APPROVED** [review]

**Comment by hardik-kapoor** [comment]
Thanks for your contribution, awarded 20 bonus points for you contributions to ag_suggest and ag_sklearn.
